topi/include/topi/cuda/nn.h

```diff
@@ -0,0 +1,107 @@
+/*!
+*  Copyright (c) 2018 by Contributors
+* \file cuda/nn.h
+* \brief CUDA schedule for lrn and l2 normalization operations
+*/
+#ifndef TOPI_CUDA_NN_H_
+#define TOPI_CUDA_NN_H_
+
+#include "tvm/tvm.h"
+#include "tvm/build_module.h"
+#include "topi/tags.h"
+
+namespace topi {
+using namespace tvm;
+namespace cuda {
+/*!
+* \brief Create a CUDA schedule for LRN
+*
+* \param target The target to generate a schedule for.
+* \param outs The output tensors.
+*
+* \return A schedule for the given ops.
+*/
+inline Schedule schedule_lrn(const Target &target, const Array<Tensor>& outs) {
+  Array<Operation> out_ops;
+  for (auto t : outs) {
+    out_ops.push_back(t->op);
+  }
+  auto s = create_schedule(out_ops);
+  auto num_thread = 64;
+  auto block_x = tvm::thread_axis(Range(), "blockIdx.x");
+  auto thread_x = tvm::thread_axis(Range(0, num_thread), "threadIdx.x");
+  auto lrn = outs[0];
+  auto sqr_sum_up = lrn->op->InputTensors()[1];
+  auto sqr_sum = sqr_sum_up->op->InputTensors()[0];
+  auto set_pad = sqr_sum->op->InputTensors()[0];
```

Comment:
I think it's better to use concrete types:
  https://github.com/dmlc/tvm/blob/master/docs/contribute/code_guide.rst#c-code-styles